python/pyarrow/tests/test_flight.py

```diff
@@ -883,8 +887,10 @@ def test_repr():
         "schema= "
         "descriptor=<pyarrow.flight.FlightDescriptor path=[]> "
         "endpoints=[] "
-        "total_records=-1 "
-        "total_bytes=-1>")
+        "total_records=1 "
+        "total_bytes=42 "
+        "ordered=true "
+        "app_metadata='7465737420617070206d65746164617461'>")
```

Comment:
This behaviour of using the hex representation might be a little confusing. It might be best to stick with the default bytes repr, ie. `app_metadata=b'test app metadata'`. This will still handle non-ascii bytes ok by displaying them like `\x00`.

Although I'm also a bit concerned that this field could be quite large, maybe we just want something like `app_metadata=<bytes with length 17>`

Reply:
This is what the c++ implementation does. I guess they use hex because it is opaque metadata. I'd leave this to / flag this with the arrow devs.

Reply:
OK yeah matching the C++ behaviour makes sense then.